product_operating_unit/models/product_category.py

```diff
@@ -0,0 +1,30 @@
+# Copyright (C) 2019 Open Source Integrators
+# Copyright (C) 2019 Serpent Consulting Services
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).
+from odoo import fields, models
+from odoo.models import Command
+
+
+class ProductCategory(models.Model):
+    _inherit = "product.category"
+
+    operating_unit_ids = fields.Many2many(
+        "operating.unit",
+        "product_category_operating_unit_rel",
+        string="Operating Units",
+    )
+
+    def write(self, vals):
+        res = super().write(vals)
+        product_template_obj = self.env["product.template"]
+        if vals.get("operating_unit_ids"):
+            for rec in self:
+                products = product_template_obj.search(
+                    [("categ_id", "child_of", rec.id)]
+                )
+                category_ou_ids = rec.operating_unit_ids.ids
+                for product in products:
+                    ou_ids = product.operating_unit_ids.ids
+                    ou_ids.extend(category_ou_ids)
+                    product.operating_unit_ids = [Command.set(ou_ids)]
```

Comment:
Nitpicking: might as well use records directly, instead of ids

Reply:
at least it's worth it to try if it fails on this

Reply:
The logic updated with records instead of ids. Thanks for the review 👍